thirtyinch/src/test/java/net/grandcentrix/thirtyinch/TiLifecycleObserverTest.java

```diff
@@ -249,6 +256,33 @@ public void onChange(final TiPresenter.State state,
     }
 
     @Test
+    public void testRemoveOtherObserver() throws Exception {
+        mPresenter.create();
+
+        final TiLifecycleObserver observer1 = mock(TiLifecycleObserver.class);
+        mPresenter.addLifecycleObserver(observer1);
+        final TiLifecycleObserver observer2 = mock(TiLifecycleObserver.class);
+        final Removable removable = mPresenter.addLifecycleObserver(observer2);
+
+        doAnswer(new Answer() {
+            @Override
+            public Object answer(final InvocationOnMock invocation) throws Throwable {
+                removable.remove();
+                return null;
+            }
+        }).when(observer1).onChange(any(TiPresenter.State.class), anyBoolean());
+
+        mPresenter.attachView(mock(TiView.class));
+
+        //receives both events
+        verify(observer1, times(2)).onChange(any(TiPresenter.State.class), anyBoolean());
+
+        // only gets first event, already unregistered when second event fires
```

Comment:
Maybe we have to adjust the comment then. `Only gets the init event. Aldready unregistered for attach event`...